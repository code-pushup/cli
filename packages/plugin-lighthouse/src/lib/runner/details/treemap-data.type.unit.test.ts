import { parseTreemapDataToBasicTrees } from './treemap-data.type.js';

describe('parseTreemapDataToBasicTrees', () => {
  it('should convert root nodes to basic trees', () => {
    expect(
      parseTreemapDataToBasicTrees({
        type: 'treemap-data',
        nodes: [
          {
            name: 'https://github.githubassets.com/assets/environment-7b93e0f0c8ff.js',
            resourceBytes: 1438,
            unusedBytes: 376,
            children: [
              {
                name: 'ui/packages',
                resourceBytes: 881,
                unusedBytes: 376,
                children: [
                  {
                    name: 'failbot/failbot-error.ts',
                    resourceBytes: 237,
                    unusedBytes: 113,
                  },
                  {
                    name: 'remove-child-patch/remove-child-patch.ts',
                    resourceBytes: 268,
                  },
                  {
                    name: 'fetch-overrides/fetch-overrides.ts',
                    resourceBytes: 376,
                    unusedBytes: 263,
                  },
                ],
              },
              {
                name: 'app/assets/modules/environment.ts',
                resourceBytes: 42,
              },
              {
                name: '(unmapped)',
                resourceBytes: 515,
              },
            ],
          },
          {
            name: 'https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-fb43816ab83c.js',
            resourceBytes: 11_903,
            unusedBytes: 10_362,
            children: [
              {
                name: 'app/assets/modules/github/behaviors',
                resourceBytes: 11_369,
                unusedBytes: 10_297,
                children: [
                  {
                    name: 'commenting/edit.ts',
                    resourceBytes: 9290,
                    unusedBytes: 8218,
                  },
                  {
                    name: 'html-validation.ts',
                    resourceBytes: 2079,
                    unusedBytes: 2079,
                    duplicatedNormalizedModuleName:
                      'app/assets/modules/github/behaviors/html-validation.ts',
                  },
                ],
              },
              {
                name: '(unmapped)',
                resourceBytes: 534,
                unusedBytes: 65,
              },
            ],
          },
        ],
      }),
    ).toMatchSnapshot();
  });
});
