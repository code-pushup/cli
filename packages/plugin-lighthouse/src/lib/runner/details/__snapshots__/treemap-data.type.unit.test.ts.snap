// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseTreemapDataToBasicTrees > should convert root nodes to basic trees 1`] = `
[
  {
    "root": {
      "children": [
        {
          "children": [
            {
              "name": "failbot/failbot-error.ts",
              "values": {
                "resourceBytes": "237 B",
                "unusedBytes": "113 B",
              },
            },
            {
              "name": "remove-child-patch/remove-child-patch.ts",
              "values": {
                "resourceBytes": "268 B",
              },
            },
            {
              "name": "fetch-overrides/fetch-overrides.ts",
              "values": {
                "resourceBytes": "376 B",
                "unusedBytes": "263 B",
              },
            },
          ],
          "name": "ui/packages",
          "values": {
            "resourceBytes": "881 B",
            "unusedBytes": "376 B",
          },
        },
        {
          "name": "app/assets/modules/environment.ts",
          "values": {
            "resourceBytes": "42 B",
          },
        },
        {
          "name": "(unmapped)",
          "values": {
            "resourceBytes": "515 B",
          },
        },
      ],
      "name": "https://github.githubassets.com/assets/environment-7b93e0f0c8ff.js",
      "values": {
        "resourceBytes": "1.4 kB",
        "unusedBytes": "376 B",
      },
    },
    "type": "basic",
  },
  {
    "root": {
      "children": [
        {
          "children": [
            {
              "name": "commenting/edit.ts",
              "values": {
                "resourceBytes": "9.07 kB",
                "unusedBytes": "8.03 kB",
              },
            },
            {
              "name": "html-validation.ts",
              "values": {
                "duplicatedNormalizedModuleName": "app/assets/modules/github/behaviors/html-validation.ts",
                "resourceBytes": "2.03 kB",
                "unusedBytes": "2.03 kB",
              },
            },
          ],
          "name": "app/assets/modules/github/behaviors",
          "values": {
            "resourceBytes": "11.1 kB",
            "unusedBytes": "10.06 kB",
          },
        },
        {
          "name": "(unmapped)",
          "values": {
            "resourceBytes": "534 B",
            "unusedBytes": "65 B",
          },
        },
      ],
      "name": "https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-fb43816ab83c.js",
      "values": {
        "resourceBytes": "11.62 kB",
        "unusedBytes": "10.12 kB",
      },
    },
    "type": "basic",
  },
]
`;
